compiler/testData/codegen/box/operatorConventions/assignOperatorOverload.kt

```diff
@@ -0,0 +1,54 @@
+import java.lang.RuntimeException
+
+// TARGET_BACKEND: JVM_IR
+// IGNORE_BACKEND: JVM_IR
+
+var result: String = "Fail"
+
+operator fun Int.assign(other: String) {
+    result = other
+}
+operator fun Container.assign(other: Int) {
+    this.value = "OK"
+}
+operator fun Container.set(i: Int, v: Int) {
+    this.value = "OK.Container.set1"
+}
+operator fun Container.set(i: Int, v: Long) {
+    this.value = "OK.Container.set2"
+}
+
+data class Foo(val x: Container)
+data class Container(var value: String)
+
+var nullCheckResult: String = "OK"
+data class NullCheck(val x: NullCheckContainer)
+data class NullCheckContainer(var value: String)
+operator fun NullCheckContainer.assign(value: String) {
+    nullCheckResult = value
+}
+
+fun box(): String {
+    // Test simple assign for local variable
+    val x = 10
+    x = "OK"
+    if (result != "OK") return "Fail: $result"
+
+    // Test simple assign for property
+    val foo = Foo(Container("Fail"))
+    foo.x = 42
+    if (foo.x.value != "OK") return "Fail: ${foo.x.value}"
+
+    // Test set() has priority
+    foo.x[1] = 2
```

Comment:
To check it correctly, we should have an alternative between `foo.x.get(1).assign(2)` and `foo.x.set(1, 2)`. To provide it, `Container` should have `get` and its result should have `assign`.

Reply:
Also, should an alternative like `x.get(1).assign(2)` be able to work, or we forbid this variant? In both cases, it would be great to have a test for this case (without `x.set` available).

Reply:
Added tests to check that `x[1] = 2` and `x.get(1) = 2` don't work. 

While `x[1] = 2` should never work, since it is ambiguous with `.set()`,  `x.get(1) = 2` might not be ambiguous. But currently it doesn't work.